cnd/src/swap_protocols/facade.rs

```diff
@@ -169,67 +167,73 @@ where
     }
 }
 
+#[async_trait::async_trait]
 impl<S> HtlcEvents<Bitcoin, Amount> for Facade<S>
 where
     S: Send + Sync + 'static,
 {
-    fn htlc_deployed(
+    async fn htlc_deployed(
         &self,
         htlc_params: HtlcParams<Bitcoin, Amount>,
-    ) -> Box<DeployedFuture<Bitcoin>> {
-        self.bitcoin_connector.htlc_deployed(htlc_params)
+    ) -> Result<Deployed<Bitcoin>, rfc003::Error> {
+        self.bitcoin_connector.htlc_deployed(htlc_params).await
     }
 
-    fn htlc_funded(
+    async fn htlc_funded(
         &self,
         htlc_params: HtlcParams<Bitcoin, Amount>,
         htlc_deployment: &Deployed<Bitcoin>,
-    ) -> Box<FundedFuture<Bitcoin, Amount>> {
+    ) -> Result<Funded<Bitcoin, Amount>, rfc003::Error> {
```

Comment:
This line highlights some non-uniformity we currently have.  We could be returning an `anyhow::Result` here.  `rcf003::Error` is implemented using `thiserror` now so implements `std::error::Error`.  I had a quick go but its not a two minute job.  Want me to investigate further and write up an issue?